src/main/scala/com/mwz/sonar/scala/scoverage/ScoverageSensor.scala

```diff
@@ -115,14 +115,30 @@ private[scoverage] abstract class ScoverageSensorInternal extends Sensor {
   }
 
   /** Returns the filename of the scoverage report for this module */
-  private[this] def getScoverageReportFilename(settings: Configuration): String = {
-    settings
-      .get(ScoverageSensorInternal.ScoverageReportPathPropertyKey)
-      .toOption
-      .getOrElse(
-        ScoverageSensorInternal.getDefaultScoverageReportPath(Scala.getScalaVersion(settings))
-      )
-  }
+  private[this] def getScoverageReportFilename(settings: Configuration): String =
+    settings.get(ScoverageSensorInternal.OldScoverageReportPathPropertyKey).toOption match {
+      case Some(path) => {
+        logger.warn(
+          s"""[scoverage] The property: '${ScoverageSensorInternal.OldScoverageReportPathPropertyKey}' is deprecated,
+             |use the new property '${ScoverageSensorInternal.NewScoverageReportPathPropertyKey}' instead""".stripMargin
+        )
+        path
+      }
+      case None => {
+        settings
+          .get(ScoverageSensorInternal.NewScoverageReportPathPropertyKey)
+          .toOption
+          .getOrElse {
+            val scalaVersion = Scala.getScalaVersion(settings)
+            val default = ScoverageSensorInternal.getDefaultScoverageReportPath(scalaVersion)
+            logger.info(
+              s"""[scoverage] Missing the property: '${ScoverageSensorInternal.NewScoverageReportPathPropertyKey}',
+                 |using the default value: '${default}'""".stripMargin
+            )
+            default
+          }
+      }
+    }
```

Comment:
It might be a bit cleaner if you separate logging from the logic, e.g.:
```scala
if (settings.hasKey(ScoverageSensorInternal.OldScoverageReportPathPropertyKey)) {
  logger.warn(
    s"""[scoverage] The property: '${ScoverageSensorInternal.OldScoverageReportPathPropertyKey}' is deprecated,
       | use the new property '${ScoverageSensorInternal.NewScoverageReportPathPropertyKey}' instead.""".stripMargin
  )
} else if (settings.hasKey(ScoverageSensorInternal.NewScoverageReportPathPropertyKey)) {
  logger.info(
    s"""[scoverage] Missing '${ScoverageSensorInternal.NewScoverageReportPathPropertyKey}' property,
       | using the default value: '$default'""".stripMargin
  )
}

settings
  .get(ScoverageSensorInternal.OldScoverageReportPathPropertyKey)
  .toOption
  .getOrElse(
    settings.get(ScoverageSensorInternal.NewScoverageReportPathPropertyKey).toOption.getOrElse(default)
  )
```

Reply:
Roger.